DOC-674: k shortest path (#766)
Macrometacorp/docs: docs/graphs/graph-queries/k-shortest-paths/index.md

- **Dynamic In-Game Navigation**: k shortest path queries can be used for pathfinding in video games, offering different routes for characters to navigate through a game world, providing a more engaging gameplay experience.

### Financial Services

- **Risk Management**: k shortest path queries can identify routes of exposure to risk, considering factors like counterparty relationships, credit risk, and market risk, helping financial institutions better understand their risk profile and make more informed decisions.
- **Portfolio Diversification**: k shortest path queries can analyze relationships between different investment assets to identify alternative paths for diversifying a portfolio, balancing risk and return.
- **Fraud Detection**: k shortest path queries can help identify unusual patterns in financial transactions, which may indicate potential fraud. By considering different paths between entities, investigators can better understand the relationships between suspicious transactions and trace the flow of funds.

## K Shortest Path Query in Action

> In this high-level example, there is a graph representing train stations and connections between cities. Each ellipse stands for a train station with the city name written inside of it (the vertices of the graph). Arrows represent train connections between cities (the edges of the graph), and the numbers near the arrows indicate the travel time between stations (used as edge weights).

Reviewer: ```suggestion
In this high-level example, a graph represents train stations and connections between cities. Each ellipse stands for a train station with the city name included in the station name (the vertices of the graph). Arrows represent train connections between cities (the edges of the graph), and the numbers near the arrows indicate the travel time between stations (used as edge weights).
```

![Train Connection Map](/img/train_map.png)

### Shortest Path

Assume a traveler wants to go from **Aberdeen** to **London** by train. The shortest path is expected to have the following vertices in this order:

1. Aberdeen
2. Leuchars
3. Edinburgh
4. York
5. London

The weight of this path is: 1.5 + 1.5 + 3.5 + 1.8 = **8.3**.

### Alternative Paths

Now, consider alternative paths due to the unavailability of the direct connection between York and London.

#### Path via Carlisle and Birmingham

An alternative path, which is slightly longer, goes like this:

1. Aberdeen
2. Leuchars
3. Edinburgh
4. York
5. **Carlisle**
6. **Birmingham**
7. London

The weight of this path is: 1.5 + 1.5 + 3.5 + 2.0 + 1.5 = **10.0**.

#### Path via Glasgow

> Another route goes via Glasgow. This path has seven stations, but it's quicker if the edge weights are compared:

Reviewer: ```suggestion
Another route goes via Glasgow. This path has seven stations, but it's quicker if you compare the edge weights:
```
